### PR TITLE
Yocto recipe for Kanto File Backup

### PR DESCRIPTION
**Pull request by hristobojilov**
[#23] Yocto recipe for Kanto File Backup
 
Signed-off-by: Hristo Bozhilov <Hristo.Bozhilov@bosch.io>

### CODE REVIEW

File: recipes-management/file-backup/files/config.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "logFile": "@FB_LOG_DD@/file-backup/file-backup.log",
+    "dir": "@FB_BACKUP_DD@",
+    "mode": "scoped"
```
Comment (konstantina-gramatova) [inline recipes-management/file-backup/files/config.json:4]: Must be configurable compliant with general approach via the .inc file.

**Review by konstantina-gramatova — CHANGES_REQUESTED** [review]

**Review by e-grigorov — CHANGES_REQUESTED** [review]
I cannot validate the recipe because of: eclipse-kanto/file-backup#17
As a result, I am not sure whether we need to change something.

**Comment by konstantina-gramatova** [comment]
@hristobojilov The Eclipse Kanto project is currently restructuring its license scheme - from Eclipse Public License 2.0 only to dual licensing under Eclipse Public License 2.0 **and** Apache License 2.0 (you can read more about it [here](https://projects.eclipse.org/projects/iot.kanto/reviews/restructuring-review)).

This contribution will be merged into the project with this new licensing scheme so its important for the project and the community to have your agreement on proceeding with it given these new circumstances :blush: 

Please, provide your feedback as a comment here that you, as an author of this contribution, understand and agree with the change explained above. Thank you!

**Comment by e-grigorov** [comment]
> I cannot validate the recipe because of: [eclipse-kanto/file-backup#17](https://github.com/eclipse-kanto/file-backup/issues/17) As a result, I am not sure whether we need to change something.

The backup issue has been resolved. The backup will start now but will log a message that the folder doesn't exist. In the Yocto recipe, could you please ensure that the folder exists i.e. the recipe should create it?

**Comment by hristobojilov** [comment]
> @hristobojilov The Eclipse Kanto project is currently restructuring its license scheme - from Eclipse Public License 2.0 only to dual licensing under Eclipse Public License 2.0 **and** Apache License 2.0 (you can read more about it [here](https://projects.eclipse.org/projects/iot.kanto/reviews/restructuring-review)).
> 
> This contribution will be merged into the project with this new licensing scheme so its important for the project and the community to have your agreement on proceeding with it given these new circumstances 😊
> 
> Please, provide your feedback as a comment here that you, as an author of this contribution, understand and agree with the change explained above. Thank you!

I understand and agree with the license change explained above.

File: recipes-management/file-backup/file-backup.inc [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+FB_CFG_DD ?= "${sysconfdir}"
+# The destination directory where the generated service file will be placed
+FB_SYSUNIT_DD ?= "${systemd_unitdir}/system"
+# The destination directory where the file backup binaries will be placed
+FB_BIN_DD ?= "${bindir}"
+# The destination directory where the file backup logs will be placed
+FB_LOG_DD ?= "${localstatedir}/log"
+# The directory to be backed up
+FB_BACKUP_DD ?= "${localstatedir}/tmp/file-backup"
+# The access mode for the the directory to be backed up or restored. Allowed values are: strict,scoped and lax.
```
Comment (e-grigorov) [inline recipes-management/file-backup/file-backup.inc:12]: - The access mode for the the directory -> The access mode for the directory
- strict,scoped -> strict, scoped

File: recipes-management/file-backup/file-backup_git.bb [hunk 3]
```diff
@@ -0,0 +1,68 @@
+DESCRIPTION = "Eclipse Kanto - File Backup"
+
+LICENSE = "EPL-2.0"
+LIC_FILES_CHKSUM = "file://src/${GO_IMPORT}/LICENSE;md5=c7cc8aa73fb5717f8291fcec5ce9ed6c"
```
Comment (e-grigorov) [inline recipes-management/file-backup/file-backup_git.bb:4]: The project license has been changed and the checksum was updated to: b62cb083c6991f5a8b6f13bf6358094d

**Review by e-grigorov — CHANGES_REQUESTED** [review]

**Review by e-grigorov — APPROVED** [review]

**Review by konstantina-gramatova — APPROVED** [review]

**Pull request merged**